velox/functions/prestosql/tests/ArrayAnyMatchTest.cpp

```diff
@@ -0,0 +1,173 @@
+ */
+
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+
+class ArrayAnyMatchTest : public functions::test::FunctionBaseTest {
+ protected:
+  // Evaluate an expression.
+  void testExpr(
+      const VectorPtr& expected,
+      const std::string& expression,
+      const VectorPtr& input) {
+    auto result = evaluate(expression, makeRowVector({input}));
+    assertEqualVectors(expected, result);
+  }
+
+  void testExpr(
+      const VectorPtr& expected,
+      const std::string& expression,
+      const RowVectorPtr& input) {
+    auto result = evaluate(expression, (input));
+    assertEqualVectors(expected, result);
+  }
+};
+
+TEST_F(ArrayAnyMatchTest, basic) {
+  auto input = makeNullableArrayVector<int64_t>(
+      {{std::nullopt, 2, 3},
+       {-1, 3},
+       {2, 3},
+       {},
+       {std::nullopt, std::nullopt}});
+  auto expectedResult =
+      makeNullableFlatVector<bool>({true, true, true, false, std::nullopt});
+  testExpr(expectedResult, "any_match(c0, x -> (x > 1))", input);
+  expectedResult = makeFlatVector<bool>({true, false, false, false, true});
+  testExpr(expectedResult, "any_match(c0, x -> (x is null))", input);
+
+  input = makeNullableArrayVector<bool>(
+      {{false, true},
+       {false, false},
+       {std::nullopt, true},
+       {std::nullopt, false}});
+  expectedResult =
+      makeNullableFlatVector<bool>({true, false, true, std::nullopt});
+  testExpr(expectedResult, "any_match(c0, x -> x)", input);
+
+  auto emptyInput = makeArrayVector<int32_t>({{}});
+  expectedResult = makeFlatVector<bool>(std::vector<bool>{false});
+  testExpr(expectedResult, "any_match(c0, x -> (x > 1))", emptyInput);
+  testExpr(expectedResult, "any_match(c0, x -> (x <= 1))", emptyInput);
+}
+
+TEST_F(ArrayAnyMatchTest, complexTypes) {
+  auto baseVector =
+      makeArrayVector<int64_t>({{1, 2, 3}, {2, 2}, {3, 3}, {4, 4}, {5, 5}, {}});
+  // Create an array of array vector using above base vector using offsets.
+  // [
+  //  [[1, 2, 3], []],
+  //  [[2, 2], [3, 3], [4, 4], [5, 5]],
+  //  [[], []]
+  // ]
+  auto arrayOfArrays = makeArrayVector({0, 1, 5}, baseVector);
+  auto expectedResult = makeNullableFlatVector<bool>({true, true, false});
+  testExpr(
+      expectedResult,
+      "any_match(c0, x -> (cardinality(x) > 0))",
+      arrayOfArrays);
+
+  // Create an array of array vector using above base vector using offsets.
+  // [
+  //  [[1, 2, 3]],  cardinalities is 3
+  //  [[2, 2], [3, 3], [4, 4], [5, 5]], all cardinalities is 2
+  //  [[]],
+  //  null
+  // ]
+  arrayOfArrays = makeArrayVector({0, 1, 5, 6}, baseVector, {3});
+  expectedResult =
+      makeNullableFlatVector<bool>({true, false, false, std::nullopt});
+  testExpr(
+      expectedResult,
+      "any_match(c0, x -> (cardinality(x) > 2))",
+      arrayOfArrays);
+}
+
+TEST_F(ArrayAnyMatchTest, bigints) {
```

Comment:
Looks like this is already covered by 'basic' test.

Reply:
Removed